### PR TITLE
feat: Fix issues 2651 2656

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
- Prove errors with granting ML privileges (references: #2651):
  - currently the test only
  - known issues section added to the docs
  - the case will be followed up on Snowflake side
- Add possibility to grant privileges to STREAMLIT (references #2656):
  - moved the validations to SDK
  - descriptions made more general
  - added tests inside SDK to prove that given streamlit an ALL works, but the FUTURE results in error
  - added tests on resources side (no tests added in old snowflake_grant_privilege_to_role resource)
  - split ALL and FUTURE to have correct validation and description
  -  the case will be followed up on Snowflake side

### CODE REVIEW

File: pkg/resources/grant_privileges_to_account_role.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ var grantPrivilegesToAccountRoleSchema = map[string]*schema.Schema{
 					Description: "Configures the privilege to be granted on all objects in either a database or schema.",
```
Comment (sfc-gh-jcieslak) [inline pkg/resources/grant_privileges_to_account_role.go:246]: I guess this should say "on future". Not related to the pr, but yeah 😅.

File: templates/resources/grant_privileges_to_account_role.md.tmpl [hunk 2]
```diff
@@ -29,6 +29,9 @@ description: |-
 
 {{ .SchemaMarkdown | trimspace }}
 
+## Known limitations
+- Setting the `CREATE SNOWFLAKE.ML.ANOMALY_DETECTION` or `CREATE SNOWFLAKE.ML.FORECAST` privileges on schema results in a permadiff because of the probably incorrect Snowflake's behavior of `SHOW GRANTS ON <object_type> <object_name>`. More in the [comment](https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2651#issuecomment-2022634952).
```
Comment (sfc-gh-jcieslak) [inline templates/resources/grant_privileges_to_account_role.md.tmpl:33]: This should be also included in the docs for snowflake_grant_privileges_to_database_role

File: pkg/resources/grant_privileges_to_database_role_acceptance_test.go [hunk 3]
```diff
@@ -531,6 +533,82 @@ func TestAcc_GrantPrivilegesToDatabaseRole_OnSchemaObject_OnFuture_InDatabase(t
...
+					resource.TestCheckResourceAttr(resourceName, "on_schema_object.0.all.0.object_type_plural", string(sdk.PluralObjectTypeStreamlits)),
+					resource.TestCheckResourceAttr(resourceName, "on_schema_object.0.all.0.in_database", databaseName),
+					resource.TestCheckResourceAttr(resourceName, "with_grant_option", "false"),
+					resource.TestCheckResourceAttr(resourceName, "id", fmt.Sprintf("%s|false|false|USAGE|OnSchemaObject|OnAll|STREAMLITS|InDatabase|%s", databaseRoleName, databaseName)),
+				),
+			},
+		},
+	})
+}
+
```
Comment (sfc-gh-jcieslak) [inline pkg/resources/grant_privileges_to_database_role_acceptance_test.go:611]: MlAnomalyDetection also is touching snowflake_grant_privileges_to_database_role, so maybe we should add a similar test to the one in snowflake_grant_privileges_to_account_role?

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: docs/resources/grant_privileges_to_account_role.md [hunk 4]
```diff
@@ -302,14 +302,14 @@ Optional:
...
 - `future` (Block List, Max: 1) Configures the privilege to be granted on all objects in either a database or schema. (see [below for nested schema](#nestedblock--on_schema_object--future))
 - `object_name` (String) The fully qualified name of the object on which privileges will be granted.
-- `object_type` (String) The object type of the schema object on which privileges will be granted. Valid values are: ALERT | DYNAMIC TABLE | EVENT TABLE | FILE FORMAT | FUNCTION | PROCEDURE | SECRET | SEQUENCE | PIPE | MASKING POLICY | PASSWORD POLICY | ROW ACCESS POLICY | SESSION POLICY | TAG | STAGE | STREAM | TABLE | EXTERNAL TABLE | TASK | VIEW | MATERIALIZED VIEW | NETWORK RULE | PACKAGES POLICY | ICEBERG TABLE
+- `object_type` (String) The object type of the schema object on which privileges will be granted. Valid values are: ALERT | DYNAMIC TABLE | EVENT TABLE | FILE FORMAT | FUNCTION | PROCEDURE | SECRET | SEQUENCE | PIPE | MASKING POLICY | PASSWORD POLICY | ROW ACCESS POLICY | SESSION POLICY | TAG | STAGE | STREAM | TABLE | EXTERNAL TABLE | TASK | VIEW | MATERIALIZED VIEW | NETWORK RULE | PACKAGES POLICY | STREAMLIT | ICEBERG TABLE
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply docs/resources/grant_privileges_to_account_role.md:305]: I think there is a task SNOW-1230423 you will cover with the grant ownership :) I don't want to take your work away! :) (I don't want to inflate this PR as it targets specific issues)

File: pkg/resources/grant_privileges_to_account_role.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/grant_privileges_to_account_role.go:246]: Changed.

File: templates/resources/grant_privileges_to_account_role.md.tmpl (hunk 2 above)
Comment (sfc-gh-asawicki) [reply templates/resources/grant_privileges_to_account_role.md.tmpl:33]: Hmmm, then, I have to confirm it with the test first. I will check it out and add if this is also true there.

File: pkg/resources/grant_privileges_to_database_role_acceptance_test.go (hunk 3 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/grant_privileges_to_database_role_acceptance_test.go:611]: As above

Comment (sfc-gh-asawicki) [reply pkg/resources/grant_privileges_to_database_role_acceptance_test.go:611]: There are a few tests missing in database role, though.

File: templates/resources/grant_privileges_to_account_role.md.tmpl (hunk 2 above)
Comment (sfc-gh-asawicki) [reply templates/resources/grant_privileges_to_account_role.md.tmpl:33]: Added.

File: docs/resources/grant_privileges_to_account_role.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply docs/resources/grant_privileges_to_account_role.md:305]: You're too kind 🙇

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**